plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/naming/OverloadedVarargsMethodInspection.java

```diff
@@ -65,10 +75,55 @@ public void visitMethod(@NotNull PsiMethod method) {
                                                         : PsiSubstitutor.EMPTY;
         if (!MethodSignatureUtil.areSignaturesEqual(sameNameMethod.getSignature(substitutor),
                                                     method.getSignature(PsiSubstitutor.EMPTY))) {
+          if (ignoreInconvertibleTypes && !areConvertibleTypesWithVarArgs(method.getParameterList(),
+                                                                          sameNameMethod.getParameterList())) {
+            continue;
+          }
           registerMethodError(method, method);
           return;
         }
       }
     }
+
+    private static boolean areConvertibleTypesWithVarArgs(@NotNull PsiParameterList parameterListWithVarArgs,
+                                                          @NotNull PsiParameterList otherParameterList) {
+      PsiParameter[] parametersWithVarArgs = parameterListWithVarArgs.getParameters();
+      PsiParameter[] otherParameters = otherParameterList.getParameters();
+
+      int lengthForVarArgs = parametersWithVarArgs.length;
+      int otherLength = otherParameters.length;
+
+      //example:
+      //parameterListWithVarArgs: (Integer i1, Integer i2, String... strings)
+      //otherParameterList: (Integer i1)
+      if (lengthForVarArgs > otherLength + 1) {
+        return false;
+      }
+
+      for (int i = 0; i < otherLength; i++) {
+        PsiType type;
+        if (i < lengthForVarArgs) {
```

Comment:
I recommend converting this `if` statement into a conditional expression (using the intention/quick-fix) and then apply the "Push conditional expression inside branch".

Reply:
Thank you! Changed to a conditional expression